ladybug/hourlyplot.py

```diff
@@ -0,0 +1,424 @@
+# coding=utf-8
+from __future__ import division
+
+from .analysisperiod import AnalysisPeriod
+from .datacollection import HourlyContinuousCollection, HourlyDiscontinuousCollection
+from .graphic import GraphicContainer
+from .legend import LegendParameters
+
+from ladybug_geometry.geometry2d.pointvector import Point2D, Vector2D
+from ladybug_geometry.geometry2d.line import LineSegment2D
+from ladybug_geometry.geometry2d.polygon import Polygon2D
+from ladybug_geometry.geometry2d.polyline import Polyline2D
+from ladybug_geometry.geometry2d.mesh import Mesh2D
+from ladybug_geometry.geometry3d.pointvector import Point3D, Vector3D
+from ladybug_geometry.geometry3d.line import LineSegment3D
+from ladybug_geometry.geometry3d.plane import Plane
+from ladybug_geometry.geometry3d.polyline import Polyline3D
+from ladybug_geometry.geometry3d.mesh import Mesh3D
+
+
+class HourlyPlot(object):
+    """Module for visualization of houlry data collections.
+
+    Args:
+        data_collection: A HourlyContinuousCollection or HourlyDiscontinuousCollection
+            which will be used to generate the hourly plot.
+        legend_parameters: An optional LegendParameter object to change the display
+            of the HourlyPlot (Default: None).
+        base_point: An optional Point3D to be used as a starting point to generate
+            the geometry of the plot (Default: (0, 0, 0)).
+        x_dim: An optional number to set the X dimension of the mesh cells (Default: 1).
+        y_dim: An optional number to set the Y dimension of the mesh cells (Default: 4).
+        z_dim: An optional number to set the Z dimension of the entire chart. This
+            will be used to make the colored_mesh3d of the chart vary in the Z
+            dimension according to the data. The value input here should usually be
+            several times larger than the x_dim or y_dim in order to be noticable
+            (e.g. 100). If 0, the colored_mesh3d will simply be flat. (Default: 0).
```

Comment:
We had to make a similar decision in buttefly before and it looks like it is easier for people to give you the final desired size than calculating the cell size. At the same time the cell size approach is much better for comparative studies where you want to keep the size the same. We might want to consider a `classmethod` to give them both options.

Reply:
Particularly because the highest value in the data set is going to be different for different data types (eg. Temperature vs. Illuminance), I think the total z dimension is going to be a much better default. But I'll add a classmethod along the lines of what you suggest with a z_dim_per_unit.

Reply:
Done.